From d623736eb4b07925c1429e4345cf5953c5eb7f7f Mon Sep 17 00:00:00 2001
From: "G.h. Gao" <guanhua.gao@nxp.com>
Date: Mon, 18 Mar 2019 18:54:50 +0800
Subject: [PATCH 104/235] staging: fsl-dpaa2/eth: Remove Tx congestion check
 from dpaa2-ceetm

With the addition of BQL support, Tx congestion feature is no longer required.

Signed-off-by: Guanhua Gao <guanhua.gao@nxp.com>
Signed-off-by: Ioana Radulescu <ruxandra.radulescu@nxp.com>
[Xulin: Original patch taken from NXP LSDK-19.06.]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 .../fsl-dpaa2/ethernet/dpaa2-eth-ceetm.c      | 33 +------------------
 1 file changed, 1 insertion(+), 32 deletions(-)

diff --git a/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth-ceetm.c b/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth-ceetm.c
index 30d5d31d6137..69f0d77994e9 100644
--- a/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth-ceetm.c
+++ b/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth-ceetm.c
@@ -1,6 +1,6 @@
 // SPDX-License-Identifier: (GPL-2.0+ OR BSD-3-Clause)
 /*
- * Copyright 2017 NXP
+ * Copyright 2017-2019 NXP
  *
  */
 
@@ -132,7 +132,6 @@ static inline int dpaa2_eth_update_tx_prio(struct dpaa2_eth_priv *priv,
 					   enum update_tx_prio type)
 {
 	struct dpaa2_ceetm_qdisc *sch = qdisc_priv(cl->parent);
-	struct dpni_congestion_notification_cfg notif_cfg = {0};
 	struct dpni_tx_schedule_cfg *sched_cfg;
 	struct dpni_taildrop td = {0};
 	u8 ch_id = 0, tc_id = 0;
@@ -144,18 +143,6 @@ static inline int dpaa2_eth_update_tx_prio(struct dpaa2_eth_priv *priv,
 
 	switch (type) {
 	case DPAA2_ETH_ADD_CQ:
-		/* Disable congestion notifications */
-		notif_cfg.threshold_entry = 0;
-		notif_cfg.threshold_exit = 0;
-		err = dpni_set_congestion_notification(priv->mc_io, 0,
-						       priv->mc_token,
-						       DPNI_QUEUE_TX, tc_id,
-						       &notif_cfg);
-		if (err) {
-			netdev_err(priv->net_dev, "Error disabling congestion notifications %d\n",
-				   err);
-			return err;
-		}
 		/* Enable taildrop */
 		td.enable = 1;
 		td.units = DPNI_CONGESTION_UNIT_FRAMES;
@@ -180,24 +167,6 @@ static inline int dpaa2_eth_update_tx_prio(struct dpaa2_eth_priv *priv,
 				   err);
 			return err;
 		}
-		/* Enable congestion notifications */
-		notif_cfg.units = DPNI_CONGESTION_UNIT_BYTES;
-		notif_cfg.threshold_entry = DPAA2_ETH_TX_CONG_ENTRY_THRESH;
-		notif_cfg.threshold_exit = DPAA2_ETH_TX_CONG_EXIT_THRESH;
-		notif_cfg.message_ctx = (u64)priv;
-		notif_cfg.message_iova = priv->cscn_dma;
-		notif_cfg.notification_mode = DPNI_CONG_OPT_WRITE_MEM_ON_ENTER |
-					      DPNI_CONG_OPT_WRITE_MEM_ON_EXIT |
-					      DPNI_CONG_OPT_COHERENT_WRITE;
-		err = dpni_set_congestion_notification(priv->mc_io, 0,
-						       priv->mc_token,
-						       DPNI_QUEUE_TX, tc_id,
-						       &notif_cfg);
-		if (err) {
-			netdev_err(priv->net_dev, "Error enabling congestion notifications %d\n",
-				   err);
-			return err;
-		}
 		break;
 	}
 
-- 
2.17.1

